File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/settings/SettingsMenu.java
package com.lasthopesoftware.bluewater.settings;

import android.app.Activity;
import android.content.Intent;
import android.view.Menu;
import android.view.MenuItem;

import com.lasthopesoftware.bluewater.R;
import com.lasthopesoftware.bluewater.about.AboutActivity;

public class SettingsMenu {

	private final Activity activity;

	public SettingsMenu(Activity activity) {
		this.activity = activity;
	}

	public boolean buildSettingsMenu(Menu menu) {
		activity.getMenuInflater().inflate(R.menu.menu_settings, menu);

		return true;
	}

	public boolean handleSettingsMenuClicks(final MenuItem item) {
		if (item.getItemId() != R.id.menu_about_app) return false;

		activity.startActivity(new Intent(activity, AboutActivity.class));
		return true;
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/settings/SettingsMenu.kt
package com.lasthopesoftware.bluewater.settings

import android.app.Activity
import android.content.Intent
import android.view.Menu
import android.view.MenuItem
import com.lasthopesoftware.bluewater.R
import com.lasthopesoftware.bluewater.about.AboutActivity
import com.lasthopesoftware.resources.strings.GetStringResources

class SettingsMenu(private val activity: Activity, private val stringResources: GetStringResources) {
	fun buildSettingsMenu(menu: Menu): Boolean {
		activity.menuInflater.inflate(R.menu.menu_settings, menu)
		val menuItem = menu.findItem(R.id.menu_about_app)
		menuItem.title = stringResources.aboutTitle
		return true
	}

	fun handleSettingsMenuClicks(item: MenuItem): Boolean {
		if (item.itemId != R.id.menu_about_app) return false
		activity.startActivity(Intent(activity, AboutActivity::class.java))
		return true
	}
}
